pkg/apimachinery/patch/patch.go

```diff
@@ -155,11 +157,38 @@ func GenerateTestReplacePatch(path string, oldValue, newValue interface{}) ([]by
 	)
 }
 
-func UnmarshalPatch(patch []byte) ([]PatchOperation, error) {
-	var p []PatchOperation
-	err := json.Unmarshal(patch, &p)
+func (p *PatchSet) AddPatch(patch []byte) error {
+	var ops []PatchOperation
+	if err := json.Unmarshal(patch, &ops); err != nil {
+		return err
+	}
+	p.patches = append(p.patches, ops...)
+	return nil
+}
+
+// Unmarshal decodes the value of the first occurance of the specified operation and path. If the operation is empty, then it decodes the first
+// instance of the path.
+func (d *PatchSet) Unmarshal(path string, operation *PatchOp, obj any) error {
```

Comment:
> This new method looks pretty complex and not that safe (that `obj` is nasty). I'm not even sure what it is suppose to be.
> 

The `any` is similar to the [Unmarshal](https://pkg.go.dev/encoding/json#Unmarshal) from the json go package

> It could have been nice if the commit came with the unit tests to see the added value and usage of this. I now see it appears in the next commit, so perhaps it is worth squashing the two.
> 

Sure, I can squash it.

> The name is also a bit confusing to me: The operation is on the patchset but the result is an unmarhsal of a specific patch entry value. Sounds like the name should be something like `UnmarshalPatchValue`. But I am not 100% sure if I understood this correctly.

+1